src/storage/PatchStore.ts

```diff
@@ -0,0 +1,46 @@
+import { Conditions } from './Conditions';
+import { Patch } from '../ldp/http/Patch';
+import { PatchHandler } from './patch/PatchHandler';
+import { Representation } from '../ldp/representation/Representation';
+import { RepresentationPreferences } from '../ldp/representation/RepresentationPreferences';
+import { ResourceIdentifier } from '../ldp/representation/ResourceIdentifier';
+import { ResourceStore } from './ResourceStore';
+
+/**
+ * {@link ResourceStore} using decorator pattern for the `modifyResource` function.
+ * If the original store supports the {@link Patch}, behaviour will be identical,
+ * otherwise one of the {@link PatchHandler}s supporting the given Patch will be called instead.
+ */
+export class PatchStore implements ResourceStore {
+  private readonly source: ResourceStore;
+  private readonly patcher: PatchHandler;
+
+  public constructor(source: ResourceStore, patcher: PatchHandler) {
+    this.source = source;
+    this.patcher = patcher;
+  }
+
+  public async addResource(container: ResourceIdentifier, representation: Representation, conditions?: Conditions): Promise<ResourceIdentifier> {
+    return this.source.addResource(container, representation, conditions);
```

Comment:
I think we've written this pattern a couple of times now.
Should we have a base class `DelegatingResourceStore` or `PassthroughResourceStore` with passthrough implementations for all methods?

Reply:
Was also thinking of something like that, but was planning of looking into it when it occurred more often. This is actually the first time I think, the previous one was the locking store and that one needed to wrap every call with the locking functionality.